firefox-ios/Client/Frontend/Home/Homepage Rebuild/LabelButtonHeaderView.swift

```diff
@@ -0,0 +1,169 @@
+        static let trailingInset: CGFloat = 16
+    }
+
+    // MARK: - UIElements
+    private lazy var stackView: UIStackView = .build { stackView in
+        stackView.backgroundColor = .clear
+        stackView.spacing = UX.inBetweenSpace
+        stackView.alignment = .fill
+        stackView.distribution = .fill
+    }
+
+    lazy var titleLabel: UILabel = .build { label in
+        label.text = self.title
+        label.font = FXFontStyles.Bold.title3.scaledFont()
+        label.adjustsFontForContentSizeCategory = true
+        label.numberOfLines = 0
+    }
+
+    private lazy var moreButton: ActionButton = .build { button in
+        button.isHidden = true
+    }
+
+    // MARK: - Variables
+    var title: String? {
+        willSet(newTitle) {
+            titleLabel.text = newTitle
+        }
+    }
+
+    var notificationCenter: NotificationProtocol = NotificationCenter.default
+
+    private var stackViewLeadingConstraint: NSLayoutConstraint?
+
+    // MARK: - Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        stackView.addArrangedSubview(titleLabel)
+        stackView.addArrangedSubview(moreButton)
+        addSubview(stackView)
+
+        setupLayout()
+        setupNotifications(forObserver: self,
+                           observing: [.DynamicFontChanged])
+    }
+
+    private func setupLayout() {
+        stackView.addArrangedSubview(titleLabel)
+        stackView.addArrangedSubview(moreButton)
+        addSubview(stackView)
+
+        stackViewLeadingConstraint = stackView.leadingAnchor.constraint(equalTo: leadingAnchor,
+                                                                        constant: UX.leadingInset)
+        stackViewLeadingConstraint?.isActive = true
+
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: topAnchor),
+            stackView.trailingAnchor.constraint(equalTo: trailingAnchor,
+                                                constant: -UX.trailingInset),
+            stackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -UX.bottomSpace),
+        ])
+
+        // Setting custom values to resolve horizontal ambiguity
+        titleLabel.setContentCompressionResistancePriority(UILayoutPriority(751), for: .horizontal)
+        titleLabel.setContentHuggingPriority(UILayoutPriority(251), for: .horizontal)
+
+        adjustLayout()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        notificationCenter.removeObserver(self)
+    }
+
+    // MARK: - Helper functions
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        moreButton.isHidden = true
+        moreButton.setTitle(nil, for: .normal)
+        moreButton.accessibilityIdentifier = nil
+        titleLabel.text = nil
+        moreButton.removeTarget(nil, action: nil, for: .allEvents)
+    }
+
+    func configure(
```

Comment:
copied over from legacy view. Only updated the configure and removed `!` from constraints